### PR TITLE
Readme updates

### PR DESCRIPTION
**Pull request by MashB**
Readme updates

### CODE REVIEW

**Review by susrisha — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -5,6 +5,27 @@ the
 user like authenticating the users, issuing the access/refresh tokens, api keys, validating the permissions for the
```
Comment (rdevalap) [inline README.md:5]: this API is responsible **for** any service related to ... of the user, like , aut

File: README.md [hunk 2]
```diff
@@ -5,6 +5,27 @@ the
...
+
+```mermaid
+graph LR;
+    A(Client) --->|Authenticate| B(Gateway) 
+    B -->|Authenticate| C(Auth Service) 
+    C -->|Authenticate| D(Keycloak Service)
+    C -->|Authorize| E(TDEI Database)
+```
+
+- Auth service is responsible for Authenticating and Authorizing the HTTP requests. Also, verifies the access token,
```
Comment (rdevalap) [inline README.md:20]: access token from/of?

Comment (rdevalap) [reply README.md:20]: In the previous "readme". all the components of mermaid diagram when written down, you used quotes around them. 
Example `Data Service` etc. Here, I don't see it. Is this intended?

File: README.md [hunk 3]
```diff
@@ -5,6 +5,27 @@ the
...
+  api-key and generates the secret token
+  for intra micro-service communication.
+- Gateway service is first point of contact for clients. Every request to gateway is Authenticated & Authorized against
+  the Auth service.
+- For Authentication, Auth service will interact with keycloak service for verification.
+    - Successful authentication auth servive will return with access_token & refresh_token.
+- For Authorization, Auth service will interact directly with TDEI database.
```
Comment (rdevalap) [inline README.md:27]: What does this TDEI database have? What is stored in it? Why does it interact direct?

File: README.md [hunk 4]
```diff
@@ -5,6 +5,27 @@ the
...
+    C -->|Authorize| E(TDEI Database)
+```
+
+- Auth service is responsible for Authenticating and Authorizing the HTTP requests. Also, verifies the access token,
+  api-key and generates the secret token
+  for intra micro-service communication.
+- Gateway service is first point of contact for clients. Every request to gateway is Authenticated & Authorized against
+  the Auth service.
+- For Authentication, Auth service will interact with keycloak service for verification.
+    - Successful authentication auth servive will return with access_token & refresh_token.
```
Comment (rdevalap) [inline README.md:26]: Upon successful authentication, auth service will return access_token ... to client?

File: README.md [hunk 5]
```diff
@@ -5,6 +5,27 @@ the
...
+- For Authorization, Auth service will interact directly with TDEI database.
+
```
Comment (rdevalap) [inline README.md:28]: In the "Cloning the Project" section. the command says
`$git clone https://TDEI-UW@dev.azure.com/TDEI-UW/TDEI/_git/**gateway**`
I think that should point to **auth-n-z**

File: README.md [hunk 6]
```diff
@@ -38,7 +59,8 @@ Application secrets are not included in the code repository. Below are the instr
 
 ###### DEV
 
-Request for **developer-local-properties.yaml** file from Admin, which should be copied to below location
+Create **developer-local-properties.yaml** file under root of `resource` folder and set the application.yaml
+placeholders.
```
Comment (rdevalap) [inline README.md:64]: Didn't understand this wording under PROD:
"Secrets will be set as an environment variables on the deployment environment.". What does it mean?

File: README.md [hunk 7]
```diff
@@ -95,22 +117,6 @@ Navigate to the below link for API documentation and API playground
...
-generate the package and create the docker image. Docker image will then be deployed to Azure app services.
-
-Process Flow Diagram:
-
-![](src/main/resources/static/images/deployment-pipeline.png)
-
-Development API documentation link
+## Development API documentation link
 
 https://tdei-gateway.azurewebsites.net/swagger-ui/index.html
```
Comment (rdevalap) [inline README.md:122]: Is this URL correct? When I click on it, I see nothing. I believe there is a **dev** missing in it.

**Pull request merged**